controllers/openstackdataplane_controller.go

```diff
@@ -209,34 +222,28 @@ func (r *OpenStackDataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 		r.Log.Info("one or more roles aren't ready, requeueing")
 		return ctrl.Result{RequeueAfter: time.Second * 5}, nil
 	}
-	if instance.Spec.DeployStrategy.Deploy && len(deployErrors) == 0 {
+	if len(deployErrors) == 0 {
 		instance.Status.Deployed = true
 		r.Log.Info("Set DeploymentReadyCondition true", "instance", instance)
 		instance.Status.Conditions.Set(condition.TrueCondition(condition.DeploymentReadyCondition, condition.DeploymentReadyMessage))
 	}
 
-	// Set DeploymentReadyCondition to False if it was unknown.
-	// Handles the case where the DataPlane is created with
-	// DeployStrategy.Deploy=false.
-	if instance.Status.Conditions.IsUnknown(condition.DeploymentReadyCondition) {
-		r.Log.Info("Set DeploymentReadyCondition false")
-		instance.Status.Conditions.Set(condition.FalseCondition(condition.DeploymentReadyCondition, condition.NotRequestedReason, condition.SeverityInfo, condition.DeploymentReadyInitMessage))
-	}
-
-	// Explicitly set instance.Spec.Deploy = false
-	// We don't want another deploy triggered by any reconcile request, it should
-	// only be triggered when the user (or another controller) specifically
-	// sets it to true.
-	instance.Spec.DeployStrategy.Deploy = false
-
 	// Set DeploymentReadyCondition to False if it was unknown.
 	// Handles the case where the Node is created with
-	// DeployStrategy.Deploy=false.
+	// If the status is Unknown at this point. We will return to avoid setting the
+	// redeployProtectionAnnotaton
 	if instance.Status.Conditions.IsUnknown(condition.DeploymentReadyCondition) {
 		r.Log.Info("Set DeploymentReadyCondition false")
 		instance.Status.Conditions.Set(condition.FalseCondition(condition.DeploymentReadyCondition, condition.NotRequestedReason, condition.SeverityInfo, condition.DeploymentReadyInitMessage))
+		return ctrl.Result{}, nil
```

Comment:
We might need to implement a more comprehensive check here. We want to avoid setting the annotation unless the deployment has actually completed successfully. So I'll try returning here and see if that achieves the desired result, otherwise I might need to implement some additional logic to ensure we only set the annotation once everything has completed.